Updates Hand class. Adds subclasses for DealerHand and PlayerHand.

Jslack3SLCC opened the pull request:
Just getting the updates for Hand methods in right now and some additions to BlackjackTest just to verify scores are working properly as it deals hand and then as the player hits. As not calling Match.update yet in the main cannot state that game is over but this gives us some good initial functionality to test out the general flow of dealing, adding cards and calculating score and then if we either initialize a match for all of this to be tied to or we make the Match methods static then we can call the update score to even see gameState value.

Let me know if you have any big concerns before we talk tomorrow otherwise we can go over anything else at that point. Thanks.

Jslack3SLCC committed:
Updates Hand class. Adds subclasses for DealerHand and PlayerHand.

Commit:
Change BlackJack test to mimic game

The Hand classes look good and work correctly.  I redid the test class
to somewhat play a game.  You have to stand repeatedly and ignore the
dealer's deal state but it seems to work.

EszetelaSLCC reviewed (approved):
Cardscore calculation is working correctly through a lot of playtests.  Approved.

Commit:
Full game in test class

Redid the test class again to add a larger do while loop that allows
playing of multiple hands and hides the dealer's cards other than the
first card until the player either loses on the deal or stands.

Commit:
Clean up blackjack test

Fixed the test class to account for blackjack and not ask for player
choice as game is over.  Made dealer hit loop with waits instead of
asking player to stand repeatedly.  Cleaned up code and added comments
on each step for future reference.
